src/ott/initializers/linear/initializers.py

```diff
@@ -313,3 +332,108 @@ def body_fn(i: int, f: jnp.ndarray) -> jnp.ndarray:
     return f.at[i].set(new_f)
 
   return jax.lax.fori_loop(0, len(f), body_fn, f)
+
+
+@functools.partial(jax.jit, static_argnums=(3,))
+def subsample(
+    rng: jax.random.PRNGKey, array: jnp.ndarray, weight: jnp.ndarray,
+    subsample_n: int
+):
+  """Subsample point cloud to uniform weighted points.
+
+  Args:
+    rng: random number generator.
+    array: point cloud, array of size N x d.
+    weight: array of size N.
+    subsample_n: number of points to subsample.
+
+  Returns:
+    subsampled point cloud, array of size subsample_n x d.
+  """
+  N = array.shape[0]
+  indices = jax.random.choice(
+      key=rng, a=N, shape=(subsample_n,), replace=True, p=weight, axis=0
+  )
+  return array[indices]
+
+
+@jax.tree_util.register_pytree_node_class
+class SubsampleInitializer(DefaultInitializer):
+  """Subsample initializer :cite:`thornton2022rethinking:22`.
+
+  Subsample each point cloud, then compute closed from
+  Sinkhorn potential between subsampled approximations. Use this potential
+  to initialize Sinkhorn potentials/scalings.
+
+  Args:
+    subsample_n: number of points to subsample from each point cloud.
+  """
+
+  def __init__(
+      self,
+      subsample_n: int,
+  ):
+    super().__init__()
+
+    self.subsample_n = subsample_n
+
+  def solve_sub_problem(self, sub_x, sub_y, epsilon):
+    """Solve subproblem with  Sinkhorn.
+
+    Args:
+      sub_x: subsampled x, array of size subsample_n x d.
+      sub_y: subsampled y, array of size subsample_n x d.
+      epsilon: regularization parameter for subsampled point cloud.
+
+    Returns:
+      Sinkhorn output object.
+    """
+    from ott.solvers.linear import sinkhorn
+
+    # can extract cost matrix entries or recompute
+    sub_geom = pointcloud.PointCloud(x=sub_x, y=sub_y, epsilon=epsilon)
```

Comment:
You're not passing the `cost_fn` here.

Reply:
yeah I agree to better pass cost_fn and not assume euclidean. 

I originally considered just extracting the cost matrix rows and columns from the original geom but then there would be no cost_fn for interpolating the potentials with the entropic map

`    # can subsample and extract cost matrix but need cost_fn for entropic map

    # x_ind = jax.random.choice(
    #     key=rng_x, a=n_x, shape=(self.subsample_n,), replace=True, p=a, axis=0
    # )
    # y_ind = jax.random.choice(
    #     key=rng_y,
    #     a=n_y,
    #     shape=(self.subsample_n_y,),
    #     replace=True,
    #     p=b,
    #     axis=0
    # )

    # sub_x, sub_y = x[x_ind], y[y_ind]

    
    # batch_index_adj = 0 if ot_prob.geom.cost_matrix.ndim == 2 else 1
    # sub_cost_matrix = jnp.take(
    #     jnp.take(ot_prob.geom.cost_matrix, x_ind, axis=0 + batch_index_adj),
    #     y_ind,
    #     axis=1 + batch_index_adj
    # )
    # sub_geom = geompointetry.Geometry(cost_matrix=sub_cost_matrix)
`